cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c

```diff
@@ -400,6 +400,7 @@ int img_mgmt_impl_erase_if_needed(uint32_t off, uint32_t len)
 int
 img_mgmt_impl_swap_type(void)
 {
+#if !defined(CONFIG_IMG_MGMT_DIRECT_XIP)
```

Comment:
It is possible to distinguish test-run while app is polling the swap type as well

Reply:
Does the direct-xip support test runs?

Reply:
Not sure if I understand reasoning behind this patch.
If in direct-xip mode one can upload image to pending slot than why bother with pending flag at all in this mode?

Reply:
@sjanc In this mode application will run from the slot it has been uploaded to. The problem is that you could still set the "opposite" slot in pending mode which means it would not be possible to overwrite and the flag would never be lifted as in XIP mode it will not switch slots, example:
0) let have app_v0.0.0 and app_v0.0.1
1) the app_v0.0.0 is running
2) set the "opposite" slot to pending
3) try to upload app_v0.0.1, will fail
4) reboot
5) the app_v0.0.0 is running, the "opposite" slot is in pending state, it is not possible to overwrite it at this point anymore.

Reply:
yes, this is what I mean. Isn't the whole point of pending flag to prevent it?
it == uploading new image before current proces of testing 'previous new image' is done?